File: AD-HOC/1441/main.js
function maxOf(h) {
    let maxValue = h;
    while (h != 1) {
        h = h % 2 ? 3 * h + 1 : h / 2;
        if (h > maxValue) maxValue = h;
    }
    return maxValue;
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');
while (h = parseInt(lines.shift())) {
    console.log(maxOf(h));
}

File: AD-HOC/1441/main.c
#include <stdio.h>

int maxOf(int h) {
    int maxValue = h;
    while (h != 1) {
        h = h % 2 ? 3 * h + 1 : h / 2;
        if (h > maxValue) maxValue = h;
    }
    return maxValue;
}

int main() {
    int h;
    for (scanf("%d", &h); h != 0; scanf("%d", &h))
        printf("%d\n", maxOf(h));
    return 0;
}
